### PR TITLE
Make all StreamWriter.Null operations nops

### LINKED ISSUE #98457: Writing to `StreamWriter.Null` can cause exceptions
This surprised me as I thought a null stream was stateless and did nothing...

```C#
string longLine = new string('#', 100000);
Parallel.For(0, 25, i => StreamWriter.Null.WriteLine(longLine));
```
produces

```
Unhandled exception. System.AggregateException: One or more errors occurred. (Arithmetic operation resulted in an overflow.) (Arithmetic operation resulted in an overflow.) (Specified argument was out of the range of valid values.) (Arithmetic operation resulted in an overflow.) (Arithmetic operation resulted in an overflow.) (Arithmetic operation resulted in an overflow.) (Arithmetic operation resulted in an overflow.) (Arithmetic operation resulted in an overflow.) (Specified argument was out of the range of valid values.) (Specified argument was out of the range of valid values.) (Arithmetic operation resulted in an overflow.) (Arithmetic operation resulted in an overflow.) (Specified argument was out of the range of valid values.)
 ---> System.OverflowException: Arithmetic operation resulted in an overflow.
   at System.IO.StreamWriter.WriteLine(String value)
   at Program.<>c__DisplayClass0_0.<<Main>$>b__0(Int32 i) in D:\bugs\streaming-spmi\null\Program.cs:line 4
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.TaskReplicator.Run[TState](ReplicatableUserAction`1 action, ParallelOptions options, Boolean stopOnFirstFailure)
   at System.Threading.Tasks.Parallel.ForWorker[TLocal,TInt](TInt fromInclusive, TInt toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.ForWorker[TLocal,TInt](TInt fromInclusive, TInt toExclusive, ParallelOptions parallelOptions, Action`1 body, Action`2 bodyWithState, Func`4 bodyWithLocal, Func`1 localInit, Action`1 localFinally)
   at System.Threading.Tasks.Parallel.For(Int32 fromInclusive, Int32 toExclusive, Action`1 body)
   at Program.<Main>$(String[] args) in D:\bugs\streaming-spmi\null\Program.cs:line 4
 ---> (Inner Exception #1) System.OverflowException: Arithmetic operation resulted in an overflow.
   at System.IO.StreamWriter.WriteLine(String value)
   at Program.<>c__DisplayClass0_0.<<Main>$>b__0(Int32 i) in D:\bugs\streaming-spmi\null\Program.cs:line 4
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__DisplayClass19_0`2.<ForWorker>b__1(RangeWorker& currentWorker, Int64 timeout, Boolean& replicationDelegateYieldedBeforeCompletion)
   at System.Threading.Tasks.TaskReplicator.Replica.Execute()<---
```

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/runtime/issues/98457

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System.IO.Tests/Stream/Stream.NullTests.cs [hunk 1]
```diff
@@ -196,14 +198,93 @@ public static async Task TestCanceledNullTextReaderAsync(TextReader input)
...
+                Assert.False(sw.AutoFlush);
+                sw.AutoFlush = true;
+                Assert.False(sw.AutoFlush);
+
+                Assert.Same(Stream.Null, sw.BaseStream);
+            }
+
+            // Use some parallelism in an attempt to validate statelessness
+            string longLine = new string('#', 100_000);
+            Parallel.For(0, 25, i => output.WriteLine(longLine));
```
Comment (adamsitnik) [inline src/libraries/System.Runtime/tests/System.IO.Tests/Stream/Stream.NullTests.cs:287]: Is it going to work fine for WASM?

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you for providing the bug fix @stephentoub !

Should we backport this fix to older TFMs?

**Comment by stephentoub** [comment]
> Should we backport this fix to older TFMs?

It's been like this I believe for 20 years. :smile: I don't think a backport is necessary.

File: src/libraries/System.Runtime/tests/System.IO.Tests/Stream/Stream.NullTests.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Runtime/tests/System.IO.Tests/Stream/Stream.NullTests.cs:287]: I believe so, but it will just end up running it serially.

**Pull request merged**